src/libraries/System.Linq/src/System/Linq/AggregateBy.cs

```diff
@@ -0,0 +1,78 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace System.Linq
+{
+    public static partial class Enumerable
+    {
+        public static IEnumerable<KeyValuePair<TKey, TAccumulate>> AggregateBy<TSource, TKey, TAccumulate>(
+            this IEnumerable<TSource> source,
+            Func<TSource, TKey> keySelector,
+            TAccumulate seed,
+            Func<TAccumulate, TSource, TAccumulate> func,
+            IEqualityComparer<TKey>? keyComparer = null) where TKey : notnull
+            => AggregateBy(source, keySelector, _ => seed, func, keyComparer);
```

Comment:
I would recommend authoring a separate iterator for this overload in order to avoid incurring a seed delegate + delegate invocation costs.

Reply:
Agreed. I have created a dedicated implementation.